In [1]:
import pandas as pd
import scipy.stats as stats
import seaborn as sns
import matplotlib.pyplot as plt

Hipotesis: En esta practica se busca comparar la media de la calificación general (overall) entre las posiciones ST, CM, CB y GK para demostrar que la posición es una caracteristica que hace significativa la diferencia en el overall, esto con una prueba ANOVA

In [2]:
# Cargar datos
df = pd.read_csv(r"C:\Users\angel\OneDrive\Escritorio\OCTAVO SEM\MINERIA DE DATOS\REPOSITORIO\AOAG_MinDat035\DATASET\players_20.csv")
df = df[df['player_positions'].isin(['ST', 'CM', 'CB', 'GK'])].copy()
df['main_position'] = df['player_positions'].str.split(',').str[0]

In [3]:
# Agrupar los valores por posición
groups = [group['overall'].values for name, group in df.groupby('main_position')]

# Aplicar ANOVA
anova_result = stats.f_oneway(*groups)

print("ANOVA F:", anova_result.statistic)
print("p-value:", anova_result.pvalue)

ANOVA F: 82.35103226725758
p-value: 2.3300634617555143e-52


In [4]:
kruskal_result = stats.kruskal(*groups)

print("Kruskal-Wallis H:", kruskal_result.statistic)
print("p-value:", kruskal_result.pvalue)

Kruskal-Wallis H: 249.10459958510432
p-value: 1.02208607192066e-53


In [5]:
# Comparando dos posiciones
st = df[df['main_position'] == 'ST']['overall']
gk = df[df['main_position'] == 'GK']['overall']

t_result = stats.ttest_ind(st, gk, equal_var=False)

print("T-test t:", t_result.statistic)
print("p-value:", t_result.pvalue)

T-test t: 5.301597870968893
p-value: 1.2119617771333973e-07


Se aplicó un (ANOVA) para comparar la media de la calificación general (overall) entre las posiciones de jugador ST (delantero centro), CM (centrocampista), CB (defensa central) y GK (portero).
El resultado fue un estadístico F de 82.35 con un valor p de 2.33 × 10⁻⁵², lo que indica una diferencia altamente significativa entre los grupos.
Por lo tanto, se concluye que la posición principal influye considerablemente en la calificación general de los jugadores